### PR TITLE
264 filter by event criteria

### LINKED ISSUE #264: Filter Resources by Event Criteria
### User story
As a requestee, I can list all the available rooms that match the criteria of my event
### Narrative
Filter available rooms by -capacity, layout, etc (example attributes are listed in the excel sheet in the drive) 
### Out of scope
N/A
### Acceptance criteria

| # | Given | When | Then |
|---|-------|------|------|
|1|I am a logged in and authenticated user|I want to find rooms or resources matching my event criteria|I can filter them based on event details|

### PR DESCRIPTION
**Pull request by jonathanp16**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**
Please provide enough information so that others can review your pull request:

Explain the **details** for making this change. What existing problem does the pull request solve?

### Test plan (required)
Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

image

image
image
image

These tests cover the filtering of rooms using boolean and numeric fields, only sending the attributes that a user checks off in the advanced filters modal.

### Closing issues
List all issues this PR resolves.
- closes #264 

### Pre-Merge checklist 
Complete these items before requesting reviews
- [x] Static Analysis Rules Validated
- [x] All commits follow naming conventions 

### Sub-tasks Checklist
- [x] UI Implementation
- [x] UI Test
- [x] Backend Implementation
- [x] Backend Tests
- [ ] Integration Tests
- [ ] System Tests
- [ ] Documentation

### CODE REVIEW

File: routes/api.php [hunk 1]
```diff
@@ -13,6 +13,45 @@
...
+          $query->where('attributes->' . $key, '>=', $value);
+        }else{
+          if(is_bool($value)) {
+            $query->where('attributes->' . $key, $value);
+          }
+        }
+
+    }
+    return response()->json($query->get());
+
```
Comment (EvanDime) [inline routes/api.php:53]: This absolutely needs to be put into a controller. This is too bulky to be left in the routes file.

Comment (jonathanp16) [reply routes/api.php:53]: Do you think I should add this as a method in the rooms controller as opposed to making a whole new one?

Comment (EvanDime) [reply routes/api.php:53]: I would make a new controller. Just to keep it compartmentalized

**Comment by Alex-Hayeur** [comment]
There's a critical code smell defined by SonarCloud here
https://sonarcloud.io/project/issues?id=CSU-Booking-Platform_application&open=AXc7GY-WaqMtq6SzvUEB&pullRequest=357&resolved=false&severities=CRITICAL&sinceLeakPeriod=true

**Comment by jonathanp16** [comment]
> There's a critical code smell defined by SonarCloud here
> https://sonarcloud.io/project/issues?id=CSU-Booking-Platform_application&open=AXc7GY-WaqMtq6SzvUEB&pullRequest=357&resolved=false&severities=CRITICAL&sinceLeakPeriod=true

Fixed the code smells @Alex-Hayeur

**Review by Alex-Hayeur — COMMENTED** [review]
I'm getting the code running locally, but for the purposes of reviewing could you please take a screenshot of the new UI as well @jonathanp16 ? And post it in the PR please.

**Review by EvanDime — APPROVED** [review]

**Review by Alex-Hayeur — APPROVED** [review]
Fantastic, thanks for the screenshot. Looks good.

**Pull request merged**